Essentials/src/com/earth2me/essentials/commands/Commandtpaall.java

```diff
@@ -42,6 +45,11 @@ private void teleportAAllPlayers(final Server server, final CommandSource sender
                 continue;
             }
             try {
+            	TPARequestEvent tpaEvent = new TPARequestEvent(sender, player, true);
+                ess.getServer().getPluginManager().callEvent(tpaEvent);
+                if(tpaEvent.isCancelled()) {
+                	return; // Potentially add a message letting requester know their request was cancelled?
```

Comment:
I think that a plugin should let the user know if it cancelled the plugin and why, though EssentialsX *could* output a generic "Your teleport request was cancelled." message - I'd need to see how other EssentialsX events do this.

Reply:
Will wait for further advice on this, willing to go either way.

Reply:
Commit will include a teleport request cancelled message, unless you've got another recommendation.